golem-worker-executor-base/src/grpc.rs

```diff
@@ -418,6 +426,75 @@ impl<Ctx: WorkerCtx, Svcs: HasAll<Ctx> + UsesAllDeps<Ctx = Ctx> + Send + Sync +
         Ok(())
     }
 
+    async fn get_worker_ids(
```

Comment:
A couple of problems with this:

- It's an internal step of the fork logic, but it has a very generic name (`get_worker_ids`) in a non-fork specific place
- The validations (whether the target worker ID exists, and the account IDs match) should be in the fork implementation (in the service layer) not here in the request handler
- In the gRPC layer the only thing that should happen is getting values out of the protobuf structures and fail if something is missing

So I would move the validations to the fork service, and then just move back the remaining few lines into the `fork_internal` and delete this function.

Reply:
> The validations (whether the target worker ID exists, and the account IDs match) should be in the fork implementation (in the service layer) not here in the request handler

@vigoo  Uh! That validation was a redundant check which I forgot to delete when refactoring. The validation already exist in service layer under the function `validate_worker_forking`